File: in_Java/1360 Number of Days Between Two Dates.java
class Solution {
    public int daysBetweenDates(String date1, String date2) {
        return Math.abs(date(date2) - date(date1));
    }
    private int date(String d){
        String[] ss = d.split("-");
        int a = Integer.parseInt(ss[0]);
        int b = Integer.parseInt(ss[1]);
        int c = Integer.parseInt(ss[2]);
        int res = 0;
        for(int i = 1971; i < a; i++){
            if(isLeap(i)) res += 366;
            else res += 365;
        }
        for(int i = 1; i < b; i++){
            if(i == 1 || i == 3 || i == 5 || i == 7 || i == 8 || i == 10 || i == 12) res += 31;
            else if(i == 2 && isLeap(a)) res += 29;
            else if(i == 2) res += 28;
            else res += 30;
        }
        res += c;
        return res;
    }
    private boolean isLeap(int year){
        boolean leap = false;
        if(year % 4 == 0)
        {
            if( year % 100 == 0)
            {
                // year is divisible by 400, hence the year is a leap year
                if ( year % 400 == 0)
                    leap = true;
                else
                    leap = false;
            }
            else
                leap = true;
        }
        else
            leap = false;
        return leap;
    }
}

File: in_Python/1360 Number of Days Between Two Dates.py
class Solution(object):
    def isleap(self, y):
        return y % 400 == 0 or (y % 4 == 0 and y % 100 != 0)
    def daysBetweenDates(self, date1, date2):
        """
        :type date1: str
        :type date2: str
        :rtype: int
        """
        d1 = date1.split('-')
        d2 = date2.split('-')
        days1, days2 = 0,0
        y1 = int(d1[0])
        y2 = int(d2[0])
        m1 = int(d1[1])
        m2 = int(d2[1])
        z1 = int(d1[2])
        z2 = int(d2[2])
        for i in range(1971, y1):
            if self.isleap(i):
                days1 += 366
            else:
                days1 += 365
        for i in range(1971, y2):
            if self.isleap(i):
                days2 += 366
            else:
                days2 += 365
        odd = [1,3,5,7,8,10,12]
        even = [4,6,9,11]
        for i in range(m1):
            if i in odd:
                days1 += 31
            elif i in even:
                days1 += 30
            elif i == 2:
                if self.isleap(y1):
                    days1 += 29
                else:
                    days1 += 28
        for i in range(m2):
            if i in odd:
                days2 += 31
            elif i in even:
                days2 += 30
            elif i == 2:
                if self.isleap(y2):
                    days2 += 29
                else:
                    days2 += 28
        days1 += z1
        days2 += z2
        return abs(days1 - days2)